### DISCUSSION TITLE
Generation of specific orientation

### DISCUSSION THREAD

**Discussion by sushdash** [neperfepx/neper#419] [Q&A] [answered]
Hello!
I want to generate INC-738 poly crystals, with specific orientations. I am trying, but I don't know whether its right or wrong. So can anyone please help me with the syntax? How can I generate 100, 110 and 111 orientations?

**Comment by rquey** [comment, marked as answer]
For a single orientation fiber, you can use `-ori fiber(...)`. For several orientation fibers, you can use `-ori 0.123*fiber(...)+0.456*fiber(...)+...` (where `0.123` and `0.456` are weighting factors); see [-ori](https://neper.info/doc/neper_t.html#cmdoption-ori).

**Reply by sushdash** [reply]
So, I used the command "neper -T -n 49 -domain 'cube(7,18,7)' -reg 1 -oricrysym cubic -ori 'fiber(0,0,1,0,0,1)' -o simulation" for generating 100 orientation. Am I right?

**Reply by rquey** [reply]
Each cell has one orientation, so this command will give you 49 cells, one orientation each, with orientations along the 001||Z fiber.

**Reply by sushdash** [reply]
So how can I generate the orientation 100, 110 and 111 (Three corners of an IPF triangle)? What should be the syntax?

**Reply by rquey** [reply]
Assuming you are plotting the Z sample direction on the IPF:

- 100||Z orientations:
```
neper -T -n 49 -domain 'cube(7,18,7)' -reg 1 -oricrysym cubic -ori "fiber(1,0,0,0,0,1)"
```

- 110||Z orientations:
```
neper -T -n 49 -domain 'cube(7,18,7)' -reg 1 -oricrysym cubic -ori "fiber(1,1,0,0,0,1)"
```

- 111||Z orientations:
```
neper -T -n 49 -domain 'cube(7,18,7)' -reg 1 -oricrysym cubic -ori "fiber(1,1,1,0,0,1)"
```

- A little bit of each, e.g.:
```
neper -T -n 49 -domain 'cube(7,18,7)' -reg 1 -oricrysym cubic -ori "0.3*fiber(1,0,0,0,0,1)+0.2*fiber(1,1,0,0,0,1)+0.5*fiber(1,1,1,0,0,1)"
```

**Reply by sushdash** [reply]
Thank you for explaining. Now I get it.
